include/yui/types.h

```diff
@@ -0,0 +1,142 @@
+#ifndef TYPES_H
+#define TYPES_H
+
+#include <imgui.h>
+
+#include <compare>
+
+namespace yui {
+
+/**
+ * @brief   Internal vec2 type
+ */
+struct vec2 {
+    constexpr vec2() = default;
+    constexpr vec2(float x, float y):
+        data{ x, y }
+    {}
+    constexpr vec2(ImVec2 obj):
+        data{ obj.x, obj.y }
+    {}
+
+    auto operator<=>(const vec2&) const = default;
+
+    inline vec2& operator+=(const vec2& obj) noexcept {
+        data[0] += obj.data[0];
+        data[1] += obj.data[1];
+        return *this;
+    }
+
+    /**
+     * @brief   Division by a scalar
```

Comment:
Either no need for the comment, or document all functions. I prefer the first one. Operators should be self explanatory.

Reply:
Will be removed.